src/bidiMapper/domains/network/NetworkStorage.spec.ts

```diff
@@ -0,0 +1,495 @@
+      });
+    });
+
+    it('should work non blocking interception ', async () => {
+      await networkStorage.addIntercept({
+        urlPatterns: [{type: 'string', pattern: 'http://not.correct.com'}],
+        phases: [Network.InterceptPhase.BeforeRequestSent],
+      });
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestPaused();
+
+      let event = await getEvent('network.beforeRequestSent');
+      expect(event).to.not.exist;
+
+      request.requestWillBeSendExtraInfo();
+      event = await getEvent('network.beforeRequestSent');
+
+      expect(event).to.deep.include({
+        isBlocked: false,
+      });
+    });
+  });
+
+  describe('network.responseStarted', () => {
+    it('should work for normal order', async () => {
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestWillBeSendExtraInfo();
+      request.responseReceived();
+      request.responseReceivedExtraInfo();
+
+      const event = await getEvent('network.responseStarted');
+      expect(event).to.exist;
+    });
+
+    it('should work for reverse order', async () => {
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestWillBeSendExtraInfo();
+
+      request.responseReceivedExtraInfo();
+      request.responseReceived();
+
+      const event = await getEvent('network.responseStarted');
+      expect(event).to.exist;
+    });
+
+    it('should work interception', async () => {
+      const interception = await networkStorage.addIntercept({
+        urlPatterns: [
+          {type: 'string', pattern: MockCdpNetworkEvents.defaultUrl},
+        ],
+        phases: [Network.InterceptPhase.ResponseStarted],
+      });
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestWillBeSendExtraInfo();
+
+      let event = await getEvent('network.responseStarted');
+      expect(event).to.not.exist;
+      request.responsePaused();
+
+      event = await getEvent('network.responseStarted');
+
+      expect(event).to.deep.include({
+        isBlocked: true,
+        intercepts: [interception],
+      });
+    });
+
+    it('should work non blocking interception ', async () => {
+      await networkStorage.addIntercept({
+        urlPatterns: [{type: 'string', pattern: 'http://not.correct.com'}],
+        phases: [Network.InterceptPhase.ResponseStarted],
+      });
+      const request = new MockCdpNetworkEvents(cdpClient);
+
+      request.requestWillBeSend();
+      request.requestWillBeSendExtraInfo();
+
+      let event = await getEvent('network.responseStarted');
+      expect(event).to.not.exist;
+
+      request.responsePaused();
+      event = await getEvent('network.responseStarted');
+      expect(event).to.not.exist;
+
+      request.responseReceived();
+      request.responseReceivedExtraInfo();
```

Comment:
For responseReceivedExtraInfo we should also have variations with the flags indicating the extra info